sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -93,25 +98,65 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
    * MSFAA received on the database with the
    * information received. If the information was already received
    * the record will not be updated.
-   * @param msfaaNumber MSFAA number
-   * @param dateSigned date in which the borrower indicated the MSFAA was signed
-   * @param serviceProviderReceivedDate date in which the MSFAA was received by/resolve from CanadaPost/Kiosk
-   * @returns update result. Only one row is supposed to be affected.
+   * @param msfaaNumber MSFAA number.
+   * @param offeringIntensity associated offering intensity.
+   * @param dateSigned date in which the borrower indicated the MSFAA was signed.
+   * @param serviceProviderReceivedDate date in which the MSFAA was received by/resolve from CanadaPost/Kiosk.
    */
-  async updateReceivedFile(
+  async updateReceivedRecord(
     msfaaNumber: string,
+    offeringIntensity: OfferingIntensity,
     dateSigned: Date,
     serviceProviderReceivedDate: Date,
-  ): Promise<UpdateResult> {
+  ): Promise<void> {
     if (!dateSigned || !serviceProviderReceivedDate) {
       throw new Error(
         "Not all required fields to update a received MSFAA record were provided.",
       );
     }
-
-    return this.repo.update(
+    // Check if the msfaa number needs to be reactivated or simply updated with a signed date. The msfaa record will need to reactivated if it is present in the database as cancelled.
+    const retrievedMSFAARecord = await this.repo.findOne({
+      select: {
+        id: true,
+        student: { id: true },
+        referenceApplication: { id: true },
+        cancelledDate: true,
+        dateSigned: true,
+        offeringIntensity: true,
+      },
+      relations: {
+        student: true,
+        referenceApplication: true,
+      },
+      where: {
+        msfaaNumber,
+        offeringIntensity,
+      },
+    });
+    if (!retrievedMSFAARecord) {
+      throw new Error(
+        `MSFAA number not found for: ${msfaaNumber} and offering intensity: ${offeringIntensity}`,
+      );
+    }
+    const systemUser = await this.systemUsersService.systemUser();
+    // If there is a retrievedMSFAARecord, check if the msfaa number was cancelled earlier. If yes, reactivate it.
+    if (retrievedMSFAARecord.cancelledDate) {
+      // Re-associate all disbursements pending e-cert generation for the same offering intensity with this msfaa number.
+      await this.msfaaNumberSharedService.reactivateMSFAANumber(
+        retrievedMSFAARecord.student?.id,
+        retrievedMSFAARecord.referenceApplication?.id,
+        retrievedMSFAARecord.offeringIntensity,
+        systemUser?.id,
+        { id: retrievedMSFAARecord.id, msfaaNumber } as MSFAANumber,
```

Comment:
do we need `as MSFAANumber`

Reply:
Yes, because otherwise typescript complains about the different properties of `newActiveMSFAA` like `newActiveMSFAA.student` in the `if` block as being non-existent.

Reply:
i guess u removed it, since its not needed